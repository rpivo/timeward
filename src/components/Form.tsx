import React from 'react';
import { InputProps } from '@components/Input';
import { StyledForm } from '@styles/components/Form.styled';

type FormProps = {
  children: React.ReactNode;
  handleButtonClick: (label: string) => void;
  isSignup: boolean;
  onSubmit: (event: React.FormEvent<HTMLInputElement>) => void;
};

type FormComposition = { Input?: React.FC<InputProps> };

const Form: React.FC<FormProps> & FormComposition =
  ({ children, onSubmit, handleButtonClick, isSignup }: FormProps): JSX.Element =>
    <StyledForm>
      <div className='button-wrapper'>
        <div
          className={`button login ${!isSignup ? 'active' : ''}`}
          onClick={(): void => handleButtonClick('login')}
          role='button'
        >
          LOGIN
        </div>
        <div
          className={`button signup ${isSignup ? 'active' : ''}`}
          onClick={(): void => handleButtonClick('signup')}
          role='button'
        >
          SIGN UP
        </div>
      </div>
      {children}
      <input onClick={onSubmit} type='submit' value={isSignup ? 'SIGN UP' : 'LOGIN'} />
    </StyledForm>;

export default Form;
