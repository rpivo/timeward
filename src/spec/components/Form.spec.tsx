import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Form from '@components/Form';

describe('Form', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Form handleLabelClick={jest.fn()} isSignup={true} onSubmit={jest.fn()}>
          <input></input>
        </Form>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('labels', () => {
    it('should call handleLabelClick when either of the labels is clicked', () => {
      const handleLabelClick = jest.fn();
      const wrapper = mount(
        <Form handleLabelClick={handleLabelClick} isSignup={true} onSubmit={jest.fn()}>
          <input></input>
        </Form>
      );
      wrapper.find('label').at(0).simulate('click');
      wrapper.find('label').at(1).simulate('click');
      expect(handleLabelClick).toHaveBeenCalledTimes(2);
    });
  });
});
