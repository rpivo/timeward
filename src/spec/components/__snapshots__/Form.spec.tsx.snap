// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  className="sc-AxjAm tWRmG"
>
  <div
    className="button-wrapper"
  >
    <div
      className="button login "
      onClick={[Function]}
      role="button"
    >
      LOGIN
    </div>
    <div
      className="button signup active"
      onClick={[Function]}
      role="button"
    >
      SIGN UP
    </div>
  </div>
  <input />
  <input
    onClick={[MockFunction]}
    type="submit"
    value="SIGN UP"
  />
</form>
`;
